src/server/req_manager.c

```diff
@@ -1686,6 +1692,13 @@ mgr_queue_set(struct batch_request *preq)
 		rc = mgr_set_attr(pque->qu_attr, que_attr_def, QA_ATR_LAST,
 			plist, preq->rq_perm, &bad, (void *)pque,
 			ATR_ACTION_ALTER);
+		if (rc == PBSE_CANNOT_SET_ROUTE_QUE) {
```

Comment:
I dislike special case code.  I'd rather everything work within the attribute framework.  The mgr_queue_set() function should remain agnostic to what attribute is being set.  It's a generic function.  Instead of having the special case code, I'd rather see the error message be changed to something that works with the existing format.  How about "route queues are incompatible with the partition attribute".  That wouldn't seem weird having queue_type printed at the end.  It would also be kind of weird that all other rejects have the attribute name at the end and this one wouldn't.